src/App.tsx

```diff
@@ -15,35 +16,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'Sort alphabetically',
+  Length = 'Sort by length',
+}
+
+function goodInUse(
+  good: string[],
+  isReverse: boolean,
+  sortTypeFilter: string,
+): string[] {
+  const currentGoods: string[] = [...good];
+
+  if (sortTypeFilter) {
+    currentGoods.sort((item1, item2) => {
+      switch (sortTypeFilter) {
+        case SortType.Alphabet:
+          return item1.localeCompare(item2);
+        case SortType.Length:
+          return item1.length - item2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    currentGoods.reverse();
+  }
+
+  return currentGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortTypeFilter, setSortTypeFilter] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const goodsList: string[] = goodInUse(
```

Comment:
You don't need to name with `List` as we know it as it's an array. Could be `sortedGoods/preparedGoods` or something similar